### PR TITLE
Add szip documentation to dataset section of user guide

### PR DESCRIPTION
**Pull request by byrnHDF**

### CODE REVIEW

**Review by loricooperhdf — APPROVED** [review]
I don't understand the szip/AEC connection but it's not necessarily for me to know. Proofed as far as spelling/grammar.

File: src/H5Dmodule.h [hunk 1]
```diff
@@ -2733,7 +2733,119 @@ allocated if necessary.
...
+write,
+ * and read datasets compressed with Szip compression, or can only read datasets compressed with Szip.
+ * Applications use Szip by setting Szip as an optional filter when a dataset is created. If the Szip encoder
+is
+ * enabled with the HDF5 library, data is automatically compressed and decompressed with Szip during I/O. If
+only the
+ * decoder is present, the HDF5 library cannot create and write Szip-compressed datasets, but it automatically
+ * decompresses Szip-compressed data when data is read.
+ *
+ * This sample HDF5 program illustrats of the use of Szip compression with HDF5.
```
Comment (lrknox) [inline src/H5Dmodule.h]:
```suggestion
 * This sample HDF5 program illustrates the use of Szip compression with HDF5.
```

File: src/H5Dmodule.h [hunk 2]
```diff
@@ -2733,7 +2733,119 @@ allocated if necessary.
...
+ * Szip compression software, providing lossless compression of scientific data, has been provided with HDF
+ * software products as of HDF5 Release 1.6.0.  Szip is an implementation of the extended-Rice lossless
+ * compression algorithm. The Consultative Committee on Space Data Systems (CCSDS) has adopted the
+extended-Rice
+ * algorithm for international standards for space applications[1,6,7]. Szip is reported to provide fast and
+ * effective compression, specifically for the EOS data generated by the NASA Earth Observatory System
+(EOS)[1].
+ * It was originally developed at University of New Mexico (UNM) and integrated with HDF by UNM researchers
+and developers.
+ *
```
Comment (lrknox) [inline src/H5Dmodule.h:2744]: There are some lines here with just a word or two on them at the end or in the middle of a sentence.  Do those need to be reformatted, or are they somehow rendered reasonably?

Comment (byrnHDF) [reply src/H5Dmodule.h:2744]: clang format changes - i Should fix those.

Comment (byrnHDF) [reply src/H5Dmodule.h:2744]: updated

**Review by derobins — APPROVED** [review]

**Pull request merged**